src/collocated_estim.jl

```diff
@@ -0,0 +1,165 @@
+        # # Timepoints to enforce Physics
+        # dataset = Array(reduce(hcat, dataset)')
+        # t = dataset[end, :]
+        # û = dataset[1:(end - 1), :]
+
+        # ode_params = Tar.extraparams == 1 ?
+        #              θ[((length(θ) - Tar.extraparams) + 1):length(θ)][1] :
+        #              θ[((length(θ) - Tar.extraparams) + 1):length(θ)]
+
+        # if length(û[:, 1]) == 1
+        #     physsol = [f(û[:, i][1],
+        #         ode_params,
+        #         t[i])
+        #                for i in 1:length(û[1, :])]
+        # else
+        #     physsol = [f(û[:, i],
+        #         ode_params,
+        #         t[i])
+        #                for i in 1:length(û[1, :])]
+        # end
+        # #form of NN output matrix output dim x n
+        # deri_physsol = reduce(hcat, physsol)
+
+        # > for perfect deriv(basically gradient matching in case of an ODEFunction)
+        # in case of PDE or general ODE we would want to reduce residue of f(du,u,p,t)
+        # if length(û[:, 1]) == 1
+        #     deri_sol = [f(û[:, i][1],
+        #         Tar.prob.p,
+        #         t[i])
+        #                 for i in 1:length(û[1, :])]
+        # else
+        #     deri_sol = [f(û[:, i],
+        #         Tar.prob.p,
+        #         t[i])
+        #                 for i in 1:length(û[1, :])]
+        # end
+        # deri_sol = reduce(hcat, deri_sol) 
+        # deri_sol = reduce(hcat, derivatives)
+
+        # Timepoints to enforce Physics 
+        t = Tar.dataset[end]
+        u1 = Tar.dataset[2]
+        û = Tar.dataset[1]
+        # Tar(t, θ[1:(length(θ) - Tar.extraparams)])'
+        #  
+
+        nnsol = NNodederi(Tar, t, θ[1:(length(θ) - Tar.extraparams)], autodiff)
+
+        ode_params = Tar.extraparams == 1 ?
+                     θ[((length(θ) - Tar.extraparams) + 1):length(θ)][1] :
+                     θ[((length(θ) - Tar.extraparams) + 1):length(θ)]
+
+        if length(Tar.prob.u0) == 1
+            physsol = [f(û[i],
+                ode_params,
+                t[i])
+                       for i in 1:length(û[:, 1])]
+        else
+            physsol = [f([û[i], u1[i]],
+                ode_params,
+                t[i])
+                       for i in 1:length(û)]
+        end
+        #form of NN output matrix output dim x n 
+        deri_physsol = reduce(hcat, physsol)
+
+        # if length(Tar.prob.u0) == 1
+        #     nnsol = [f(û[i],
+        #         Tar.prob.p,
+        #         t[i])
+        #              for i in 1:length(û[:, 1])]
+        # else
+        #     nnsol = [f([û[i], u1[i]],
+        #         Tar.prob.p,
+        #         t[i])
+        #              for i in 1:length(û[:, 1])]
+        # end
+        # form of NN output matrix output dim x n
+        # nnsol = reduce(hcat, nnsol)
+
+        # > Instead of dataset gradients trying NN derivatives with dataset collocation 
+        # # convert to matrix as nnsol  
+
```

Comment:
these long comments should get cleaned up

Reply:
okay